### DISCUSSION TITLE
Writing a .nas mesh file

### DISCUSSION THREAD

**Discussion by scdivi** [nschloe/meshio#1387] [Q&A]
I have a triangulated information, i.e., vertices (1500 x 3) and faces (2430 x 3). I am trying to write a .nas file using meshio v5.3.4 as following:
```
msh = meshio.Mesh(vertices, [("triangle", faces)])
msh.write(fname+'.nas')
```

However, I get an assertion error as:
```
File "python_package_path\meshio\nastran\_nastran.py", line 415, in _float_to_nastran_string
    assert len(out) <= 16
AssertionError
```

I am unable to breakdown what is the cause of this error. Can anyone please help me understand it?

Thanks in advance!
Sai

**Comment by nickvazz** [comment]
@scdivi @nschloe 
Found by changing https://github.com/nschloe/meshio/blob/b2ee99842e119901349fdeee06b5bf61e01f450a/src/meshio/nastran/_nastran.py#L413
to 
```python
    out = np.format_float_scientific(value, exp_digits=1, precision=9).replace(
```
seemed to fix the problem
